go/README.md

```diff
@@ -1 +1,177 @@
+libConv "bitbucket.org/aanzeeonline/libgoaz/conv"
+libCrypt "bitbucket.org/aanzeeonline/libgoaz/crypt"
+
+Don't use a name that it meaningless, like: util, common, misc.
+There is one exception to this: helpers.
+We will use helpers to store various helper-functions and over time, we can decide to group them into
+a package that is more suitable.
+
+## Imports
+
+order of imports
+
+- packages of current module
+- packages of aan zee lib's
+- the rest
+
+## Variable declaration
+
+Top level order of declaration:
+- constants
+- types
+- variables
+
+Global / package / function
+
+Don't use many Global var's.
+
+If a package var is used in more than 1 file, put it in a base.go
+
+Declare function var's mostly at the top of the function, or close to the first usage.
+
+## Functions and methods
+
+You can house more than 1 function inside a file (inside a package).
+The file name represents the content it houses.
+
+- naming
+	- give the function a name it represent.
+	- You can simplify a function name when it's a common abbreviation.
+		db.Initialize() could be db.Init()
+- no getters (the fact it begins with a capital and returns value, makes it a getter)
+- setters can be used
+- if crud, follow crud order inside source? (thoughts about this?)
+- global vs local
+  - first all global functions and methods (they start with capitals)
+  - local function below that (they start with lowercase)
+- return var's:
+  - when returning an error, it is always the last value that returns
+
+## Testing
+
+- code coverage (100% of relevant)
+- no . imports
+
+- Handlers should only test the basic working.
+- Services should test every scenario possible and should be tested with 100% codecoverage.
+  - if there are model validations, these should be tested too.
+- Repositories should test every scenario possible and should be tested with 100% codecoverage.
+- 
+
+## Comments
+
+- package
+  - doc
+    - Describes why the package exists
+    - it has the package format:
+      - Package the_name the_rest_of_the_text.
+- section
+  - special format:
+	/* ------------------------------- Imports --------------------------- */
+	/* ------------------------------- Constants/Types/Variables --------------------------- */
+	/* ------------------------------- Methods/Functions --------------------------- */
+- variable
+  - single line
+    - format of that:
+    // explain what it does and/or why it exists
+
+- function/method
+  - multi line
+    - if the function/method is public:
+      - first line is a complete sentence, starting with capital, ending with dot.
+    - if it is private:
+      - first line is a complete sentence, starting with lowercase, not ending with dot.
+    - is written in third-person singular with minimal 4 words
+     - format of that
+	/*
+	YourFunction does something spectacular.
+	*/
+
+## Variable naming
+
+A variable inside a function can have a short name like "a", if it is clear what it is used for.
```

Comment:
Variables should always explain what it does
Like in this example:
assert := asserts.New(t)
assert.Equals("Check", someVar)

or in a loop, like in this example:
for key, user := range users {
    println("key:", key)
    println("val:", user)
}

In any other case, it is wise to name the variable after the thing it contains.
If is can hold a collection of things, name it plural, like "users".
If it can only hold one value, name it singular, like "user".
if it is a boolean var, start it with something like "has", or "is".

which is mostly used for checking if an element is present in a map.
It's also common use, when a variable name holds an abbreviation, that the letters are all capitals, or all lowercase.
like "SQLStatement" or "sqlStatement"

Reply:
Good suggestion. Let's use this instead.

Reply:
Also, what's the convention on naming const variables?

Reply:
eerste stuk overnemen